### DISCUSSION TITLE
add event handling to tracee-ebpf

### DISCUSSION THREAD

**Discussion by AsafEitani** [aquasecurity/tracee#1209] [Development]
We should add a function to tracee-ebpf in tracee.bpf.c that will handle the event submission for all events.
That way we will have one central place where configurations could be checked, artifact can be captured and enrichments can be made such as adding the syscall that called the function (detect-syscall flag).

This will make the code more organized, provide easier debugging of issues and reduce the overall duplicate code.

**Comment by yanivagman** [comment]
We have a function named events_perf_submit() used for event submission.
It is unclear to me what this sentence means:

> one central place where configurations could be checked, artifact can be captured and enrichments can be made such as adding the syscall that called the function (detect-syscall flag).

Let's convert this issue to a discussion before we decide if we want to open an issue for this
